server/client.go

```diff
@@ -685,9 +705,13 @@ func (c *client) writeLoop() {
 		c.mu.Unlock()
 
 		if isClosed {
-			return
+			break
 		}
 	}
+
+	c.mu.Lock()
+	c.out.p, c.out.s = nil, nil
```

Comment:
Can move this to where we set the isClosed bool to avoid the second lock/unlock.

Reply:
I would rather do this here once when the connection is closed, than having to check isClosed (twice) at each loop iteration, no?

Reply:
Please confirm if ok to do it at end of loop like this or want me to check in the loop.

Reply:
I would check in the loop.

Reply:
ok...